ON-37187 # Added command release product

mymattcarroll opened the pull request.

mymattcarroll committed:
ON-37187 # Added command release product

jdawg093 commented on src/bin.ts:
```diff
@@ -4,14 +4,40 @@ import path from 'path'
+
+${chalk.bold('Examples')}
+
+  oneblink-release repository
+  oneblink-release repository --no-name
+  oneblink-release repository --name="Inappropriate Release Name"
+  oneblink-release repository 1.1.1
+  oneblink-release repository 1.1.1 --cwd ../path/to/code
+  oneblink-release repository 1.1.1-uat.1 --no-git
```
```suggestion
  oneblink-release product
  oneblink-release product --name="Inappropriate Release Name"
```

jdawg093 commented on src/startProductRelease.ts:
```diff
@@ -0,0 +1,291 @@
+        continue
+      }
+
+      const unreleasedChangelogEntries = unreleasedVersion.body.trim()
+      if (!unreleasedChangelogEntries) {
+        ora(
+          `Skipping "${codeBaseName}" as CHANGELOG.md does not contain any entries in the "Unreleased" section.`,
+        )
+          .start()
+          .info()
+        continue
+      }
```
From my understanding, the dependency checker that adds to the changelog runs in `startReleaseProcess()`/at a later point, would this mean that any repo that only had dependency updates (e.g Forms-CDN, Approvals-Client, etc.) would be skipped?

divporter replied:
perhaps another way of evaluating this is checking if the last commit on the default branch is after the commit from the last release? It seems a little fragile relying on changelog entries.

mymattcarroll replied:
Yep, you are right @jdawg093.

I was trying to avoid checking commits as we have CI jobs that do commits after release commits for documentation. We may may have others in the future. This is why I deemed it fragile relying on commits and seemed more reliable to skip releases if there are no changelog entries.

I will keep thinkering.

mymattcarroll committed:
Update src/bin.ts

Co-authored-by: jdawg093 <49189518+jdawg093@users.noreply.github.com>

mymattcarroll committed:
ON-37187 # Changed product release process to always prompt for release

Include last commit in information for user to determine if repository should be released

mymattcarroll committed:
ON-37187 # Removed npm install command
